### ISSUE TITLE
Feature request: auto create aws bucket

### ISSUE THREAD

**Issue by mikemaccana** [architect/architect#485]
Given:

```
@aws
region us-west-2
profile default
runtime nodejs10.x
bucket somename
```

`arc deploy` should:
 - Check that bucket `somename` exists
 - If bucket `somename` does not exist, arc should create it in region
 - If bucket `somename` exists and is in region `us-west-2` continue 
 - If the bucket `somename` exists and is *not* in region `us-west-2` fail with a handy message saying that the `bucket` used for deploy must be in the same region.

# Why this would be useful

It's come up a few times in Slack. Also `arc deploy`, if the bucket doesn't exist, will show a message from aws saying to make the bucket using the command `aws s3 mb s3://someapp` but nearly all users will need `aws s3 mb s3://someapp --region us-west-2`

**Comment by ryanblock** [comment]
![this-up](https://user-images.githubusercontent.com/200964/66705065-fccd9a00-ecd6-11e9-9e7e-f4e98df52434.gif)

**Comment by brianleroux** [comment]
No. This will create a tonne of problems and we have enough of those already.

**Comment by brianleroux** [comment]
Sorry hasty close with low explains. This will create determinism errors, surface for an extremely big security hole, and (the worst) it is papering over an already leaking abstraction with Cloudformation. Users will complain if we create a GUID bucket every time they deploy. We could save the bucket name in a file on their system …but that creates new/different problems. 

AWS has acknowledged they will be removing this limitation so we can solve with patience rather than introducing the problems enumerated above. I don't like it either but this is a docs problem that we need users to own not us.

**Comment by ryanblock** [comment]
> This will create determinism errors

We already have already witnessed many, especially related to bucket region, and conflicts with default credentials regions (which is super subtle). Which are you thinking of?

> ... surface for an extremely big security hole

By having Architect manage the bucket, we can ensure it's super locked down, likely ensuring greater security than a quick slapdash setup that folks may promptly forget about it.

> Users will complain if we create a GUID bucket every time they deploy.

I think we'd just create a single bucket one time if one isn't present, not a bucket for each deploy.

> We could save the bucket name in a file on their system …but that creates new/different problems.

I imagine we'd just write that value to their .arc file and be done with it. Parser was recently updated to be able to write back out to disk, so that seems pretty straightforward too.

> AWS has acknowledged they will be removing this limitation so we can solve with patience

This is definitely true!

>  I don't like it either but this is a docs problem that we need users to own not us.

For all the reasons above, I disagree. But there may be considerations or subtleties I'm not considering, please do dig deeper!

**Comment by brianleroux** [comment]
We need a much better lex/token/parse (which I have started) for architect before we get into the business of manually mutating userland files. Asking you all to disagree and commit here.

**Comment by ryanblock** [comment]
After discussing further internally, we think this could be made much more viable by auto-creating the bucket and setting/getting the bucket value from SSM (which is where we store the rest of Architect's app metadata, such as env vars).

I'm not sure when we'll get to this, but hopefully it'll be significantly before AWS straight up fixes this issue for CloudFormation once and for all!

**Comment by brianleroux** [comment]
💕

**Comment by ryanblock** [comment]
Released as `6.1`!

**Comment by mikemaccana** [comment]
Brilliant thanks gents! ⭐✨
